grpctesting/testpb/test.pb.go

```diff
@@ -1,1037 +1,331 @@
-	0x1a, 0x8a, 0xee, 0x5d, 0x28, 0xee, 0x14, 0x34, 0x2d, 0x82, 0x82, 0x0c, 0xed, 0xcc, 0xa5, 0x04,
-	0x9a, 0x49, 0x4c, 0x32, 0x2d, 0x3e, 0x9f, 0x1b, 0x97, 0x3e, 0x82, 0xf4, 0x49, 0x24, 0x69, 0x85,
-	0x2e, 0xb4, 0x74, 0xd1, 0xe5, 0x3d, 0xe7, 0x70, 0x7e, 0x42, 0x00, 0x1c, 0x5a, 0x97, 0x68, 0xa3,
-	0x9c, 0xa2, 0x47, 0x13, 0xa3, 0xb3, 0x54, 0x8a, 0x3c, 0x9f, 0xe2, 0x7c, 0x64, 0x30, 0xf1, 0x9c,
-	0x1e, 0xb3, 0x06, 0xd4, 0x6e, 0xa4, 0x76, 0x6f, 0x6c, 0x0e, 0xad, 0x7b, 0x51, 0x4c, 0x38, 0xbe,
-	0x96, 0x68, 0x1d, 0xed, 0x40, 0x6d, 0x36, 0x9a, 0x96, 0x18, 0x91, 0x1e, 0x89, 0x9b, 0x7c, 0x79,
-	0x50, 0x06, 0x6d, 0x3b, 0x45, 0xd4, 0xa9, 0x13, 0x12, 0x53, 0x69, 0xa3, 0x7f, 0x3d, 0x12, 0xd7,
-	0x78, 0x2b, 0x80, 0x43, 0x21, 0xf1, 0xce, 0xd2, 0x04, 0x0e, 0xd1, 0x18, 0x65, 0xd2, 0x4c, 0xe5,
-	0x98, 0x1a, 0x74, 0xa5, 0x29, 0x30, 0x8f, 0xaa, 0x3d, 0x12, 0xb7, 0xf9, 0x41, 0xa0, 0xae, 0x55,
-	0x8e, 0x7c, 0x45, 0xb0, 0x4b, 0xd8, 0x5f, 0x06, 0x5b, 0xad, 0x0a, 0x8b, 0x3e, 0xf9, 0x71, 0x3d,
-	0x39, 0x1c, 0x34, 0x82, 0x46, 0xa6, 0xca, 0xc2, 0xa1, 0x59, 0x65, 0xfe, 0x9c, 0xe7, 0xef, 0x55,
-	0x68, 0x0d, 0xd1, 0xba, 0x01, 0x9a, 0x99, 0xc8, 0x90, 0x72, 0x68, 0x7a, 0xbf, 0xb0, 0x8a, 0x9e,
-	0x24, 0xbf, 0xef, 0x4e, 0x02, 0x7d, 0x7c, 0xf6, 0x17, 0xbd, 0xde, 0x88, 0x55, 0xe8, 0x00, 0xfe,
-	0x7b, 0x84, 0x9e, 0x6e, 0xd6, 0x87, 0xa7, 0xdb, 0xda, 0xf4, 0x61, 0x55, 0xd4, 0xbf, 0xc8, 0x76,
-	0xce, 0x9b, 0xd7, 0xb0, 0x0a, 0x7d, 0x82, 0x3d, 0xaf, 0xbf, 0x15, 0xd6, 0xed, 0xb4, 0x6b, 0x9f,
-	0xd0, 0x17, 0x00, 0x8f, 0x0d, 0x9c, 0xc1, 0x91, 0xdc, 0xa9, 0x79, 0x4c, 0xfa, 0xe4, 0xaa, 0xf3,
-	0xb1, 0xe8, 0x92, 0xcf, 0x45, 0x97, 0x7c, 0x2d, 0xba, 0xe4, 0xb9, 0xbe, 0x14, 0x8e, 0xeb, 0xe1,
-	0xf3, 0x5e, 0x7c, 0x07, 0x00, 0x00, 0xff, 0xff, 0x92, 0x21, 0xeb, 0x3f, 0xca, 0x02, 0x00, 0x00,
-}
-
-// Reference imports to suppress errors if they are not otherwise used.
-var _ context.Context
-var _ grpc.ClientConn
-
-// This is a compile-time assertion to ensure that this generated file
-// is compatible with the grpc package it is being compiled against.
-const _ = grpc.SupportPackageIsVersion4
-
-// TestServiceClient is the client API for TestService service.
-//
-// For semantics around ctx use and closing/ending streaming RPCs, please refer to https://godoc.org/google.golang.org/grpc#ClientConn.NewStream.
-type TestServiceClient interface {
-	PingEmpty(ctx context.Context, in *Empty, opts ...grpc.CallOption) (*PingResponse, error)
-	Ping(ctx context.Context, in *PingRequest, opts ...grpc.CallOption) (*PingResponse, error)
-	PingError(ctx context.Context, in *PingRequest, opts ...grpc.CallOption) (*Empty, error)
-	PingList(ctx context.Context, in *PingRequest, opts ...grpc.CallOption) (TestService_PingListClient, error)
-	PingStream(ctx context.Context, opts ...grpc.CallOption) (TestService_PingStreamClient, error)
-}
-
-type testServiceClient struct {
-	cc *grpc.ClientConn
-}
-
-func NewTestServiceClient(cc *grpc.ClientConn) TestServiceClient {
```

Comment:
Current blocker: somehow new genproto that uses just protobuf (not gogo) does not build me grpc ):

Reply:
```
OLDPATH=${PATH}

mkdir -p /tmp/protobin/
cp ${PROTOC_GEN_GO_BIN} /tmp/protobin/protoc-gen-go
cp ${PROTOC_GEN_GO_GRPC_BIN} /tmp/protobin/protoc-gen-go-grpc
PATH=${OLDPATH}:/tmp/protobin

DIRS="grpctesting/testpb"
echo "generating protobuf code"
for dir in ${DIRS}; do
	pushd ${dir}
		${PROTOC_BIN} --go-grpc_out=.\
      -I=. \
			*.proto

			${GOIMPORTS_BIN} -w *.pb.go
	popd
done
```
Anything obvious @johanbrandhorst :thinking:

Reply:
```
var file_test_proto_depIdxs = []int32{
	0, // 0: grpc_middleware.testpb.TestService.PingEmpty:input_type -> grpc_middleware.testpb.Empty
	1, // 1: grpc_middleware.testpb.TestService.Ping:input_type -> grpc_middleware.testpb.PingRequest
	1, // 2: grpc_middleware.testpb.TestService.PingError:input_type -> grpc_middleware.testpb.PingRequest
	1, // 3: grpc_middleware.testpb.TestService.PingList:input_type -> grpc_middleware.testpb.PingRequest
	1, // 4: grpc_middleware.testpb.TestService.PingStream:input_type -> grpc_middleware.testpb.PingRequest
	2, // 5: grpc_middleware.testpb.TestService.PingEmpty:output_type -> grpc_middleware.testpb.PingResponse
	2, // 6: grpc_middleware.testpb.TestService.Ping:output_type -> grpc_middleware.testpb.PingResponse
	0, // 7: grpc_middleware.testpb.TestService.PingError:output_type -> grpc_middleware.testpb.Empty
	2, // 8: grpc_middleware.testpb.TestService.PingList:output_type -> grpc_middleware.testpb.PingResponse
	2, // 9: grpc_middleware.testpb.TestService.PingStream:output_type -> grpc_middleware.testpb.PingResponse
	5, // [5:10] is the sub-list for method output_type
	0, // [0:5] is the sub-list for method input_type
	0, // [0:0] is the sub-list for extension type_name
	0, // [0:0] is the sub-list for extension extendee
	0, // [0:0] is the sub-list for field type_name
}

func init() { file_test_proto_init() }
``` 

kind of generates something.. but that's it ;p

Reply:
protoc-gen-go v1.25.0 doesn't have grpc. You need `protoc-gen-go-grpc` or `1.4.x` or 
https://github.com/grpc/grpc-go/pull/3453

Reply:
Used `google.golang.org/grpc/cmd/protoc-gen-go-grpc v0.0.0-20200723182653-9106c3fff523`

I actually made a typo and used only go protoc gen.

BTW This was required as well: `${PROTOC_BIN} --go_out=. --go-grpc_out=.\` so somehow those files are generated separatedly ;p

Which is nice I guess (:  Thanks!